ingeniamotion/drive_tests.py

```diff
@@ -134,8 +140,27 @@ def secondary_ssi_test(self, servo=DEFAULT_SERVO, axis=DEFAULT_AXIS,
 
     def __feedback_test(self, feedback, servo=DEFAULT_SERVO, axis=DEFAULT_AXIS,
                         apply_changes=True):
-        feedbacks_test = Feedbacks(self.mc, servo, axis, feedback)
-        output = feedbacks_test.run()
+        # if feedback == SensorType.ABS1:
+        #     feedbacks_test = AbsoluteEncoder1Test(self.mc, servo, axis)
+        # elif feedback == SensorType.QEI:
+        #     feedbacks_test = DigitalIncremental1Test(self.mc, servo, axis)
+        # elif feedback == SensorType.HALLS:
+        #     feedbacks_test = DigitalHallTest(self.mc, servo, axis)
+        # elif feedback == SensorType.SSI2:
+        #     feedbacks_test = SecondarySSITest(self.mc, servo, axis)
+        # elif feedback == SensorType.BISSC2:
+        #     feedbacks_test = AbsoluteEncoder2Test(self.mc, servo, axis)
+        # elif feedback == SensorType.QEI2:
+        #     feedbacks_test = DigitalIncremental2Test(self.mc, servo, axis)
+
+        feedbacks_test = {SensorType.ABS1: AbsoluteEncoder2Test(self.mc, servo, axis),
+                          SensorType.QEI: DigitalIncremental1Test(self.mc, servo, axis),
+                          SensorType.HALLS: DigitalHallTest(self.mc, servo, axis),
+                          SensorType.SSI2: SecondarySSITest(self.mc, servo, axis),
+                          SensorType.BISSC2: AbsoluteEncoder2Test(self.mc, servo, axis),
+                          SensorType.QEI2: DigitalIncremental2Test(self.mc, servo, axis)}
```

Comment:
Don't create the class instance inside the dictionary, create it after choosing it.

Reply:
In addition, create a public function that gives the correct object initialized